test/e2e/scenario/recorder/shadowDom.scenario.ts

```diff
@@ -0,0 +1,232 @@
+        expect(mouseInteraction).toBeTruthy()
+        expect(mouseInteraction.data.source).toBe(IncrementalSource.MouseInteraction)
+        expect((mouseInteraction.data as MouseInteractionData).id).toBe(labelNode.id)
+      })
+
+    createTest('record input')
+      .withRum({ defaultPrivacyLevel: 'allow', enableExperimentalFeatures: ['recordShadowDom'] })
+      .withRumInit(initRumAndStartRecording)
+      .withSetup(bundleSetup)
+      .withBody(
+        html`
+          ${simpleShadowDom}
+          <my-open-web-component id="some" />
+        `
+      )
+      .run(async ({ serverEvents }) => {
+        await browserExecute(() => document.documentElement.outerHTML)
+        const input = await getNodeInsideShadowDom('input')
+        await input.addValue('t')
+
+        await flushEvents()
+        expect(serverEvents.sessionReplay.length).toBe(1)
+        const fullSnapshot = findFullSnapshot(getFirstSegment(serverEvents))!
+        const inputNode = findElementWithTagName(fullSnapshot.data.node, 'input')!
+        const inputRecord = findIncrementalSnapshot(getFirstSegment(serverEvents), IncrementalSource.Input)!
+        expect(inputRecord).toBeTruthy()
+        expect(inputRecord.data.source).toBe(IncrementalSource.Input)
+        expect((inputRecord.data as InputData).id).toBe(inputNode.id)
+        expect((inputRecord.data as { text: string }).text).toBe('totot')
+      })
+
+    createTest('record mutation')
+      .withRum({ defaultPrivacyLevel: 'allow', enableExperimentalFeatures: ['recordShadowDom'] })
+      .withRumInit(initRumAndStartRecording)
+      .withSetup(bundleSetup)
+      .withBody(
+        html`
+          ${simpleShadowDom}
+          <my-open-web-component id="some" />
+        `
+      )
+      .run(async ({ serverEvents }) => {
+        await browserExecute(() => document.documentElement.outerHTML)
+        const input = await getNodeInsideShadowDom('input')
+        await input.addValue('t')
+
+        await flushEvents()
+        expect(serverEvents.sessionReplay.length).toBe(1)
+        const fullSnapshot = findFullSnapshot(getFirstSegment(serverEvents))!
+
+        const pNode = findElementWithTagName(fullSnapshot.data.node, 'p')!
+        const mutationRecord = findIncrementalSnapshot(getFirstSegment(serverEvents), IncrementalSource.Mutation)!
+        expect(mutationRecord).toBeTruthy()
+        expect(mutationRecord.data.source).toBe(IncrementalSource.Mutation)
+        const mutationData = mutationRecord.data as BrowserMutationData
+        expect(mutationData.adds.length).toBe(1)
+        expect(mutationData.adds[0].parentId).toBe(pNode.id)
+        const textNode = mutationData.adds[0].node as TextNode
+        expect(textNode.type).toBe(NodeType.Text)
+        expect(textNode.textContent).toBe('totot')
+        expect(mutationData.removes.length).toBe(1)
+        expect(mutationData.removes[0].parentId).toBe(pNode.id)
+        expect(mutationData.removes[0].id).toBe(pNode.childNodes[0].id)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Use [`createMutationPayloadValidatorFromSegment`](https://github.com/DataDog/browser-sdk/blob/main/packages/rum/test/utils.ts#L476) to test for records thoroughly. This should greatly simplify those assertions. See other usages for examples how to use it.